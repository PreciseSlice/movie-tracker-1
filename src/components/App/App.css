body {
  margin: 0;
  font-family: 'Nunito Sans', sans-serif;
  text-align: center;
}

h1,
h2,
h3,
h4,
h5 {
  font-weight: 200;
}