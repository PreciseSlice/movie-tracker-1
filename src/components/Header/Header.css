nav {
  padding: 30px 0 0 0;
  background-color: #f9f7f7;
  height: 100px; }

nav a {
  text-decoration: none;
  color: #000; }

nav button {
  border: none;
  margin: 0 7px 0 0;
  background-color: white;
  height: 25px; }

.login-container {
  display: inline-block;
  position: absolute;
  top: 10px;
  right: 15px; }

.login-container p {
  margin: 0 0 5px 0;
  font-size: .8em; }

.title {
  margin: 0;
  font-size: 4em; }
