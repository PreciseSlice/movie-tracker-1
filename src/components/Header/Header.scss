nav {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #f9f7f7;
  height: 130px;
}

.title {
  margin: 0;
  font-size: 4em;
}