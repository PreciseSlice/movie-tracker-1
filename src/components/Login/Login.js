import React, { Component } from 'react';
import { connect } from 'react-redux';
// import PropTypes from 'prop-types';

import { fetchUser } from '../../helpers/helper';
import { getUser, login } from '../../actions/index';
import PropTypes from 'prop-types';

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };
  }

  async validateLogin() {
    const { handleLogin } = this.props;
    const { email, password } = this.state;
    const userFetchResponse = await fetchUser();

    const userMatch = userFetchResponse.data.find(user => {
      return user.email === email && user.password === password;
    });
    if (userMatch) {
      handleLogin(true);
    } else {
      throw new Error('Email address and password not found!');

    }
  }

  render() {
    const { handleSubmit } = this.props;
    return (
      <section>
        <form
          onSubmit={event => {
            event.preventDefault();
            handleSubmit(this.state.email.toLowerCase(), this.state.password);
            this.validateLogin();
          }}
        >
          <input
            value={this.state.email}
            placeholder="email"
            type="email"
            autoComplete="email"
            onChange={event => this.setState({ email: event.target.value })}
            required
          />
          <input
            value={this.state.password}
            placeholder="password"
            type="password"
            autoComplete="on"
            onChange={event => this.setState({ password: event.target.value })}
            required
          />
          <button>Log In</button>
        </form>
      </section>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  handleSubmit: (email, password) => dispatch(getUser(email, password)),
  handleLogin: boolean => dispatch(login(boolean))
});

Login.propTypes = {
  handleSubmit: PropTypes.func
};

export default connect(null, mapDispatchToProps)(Login);
