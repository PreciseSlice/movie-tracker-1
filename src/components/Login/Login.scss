form {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 20px 0 0 0;
}

input {
  margin: 7px 0;
  width: 40%;
  height: 30px;
  text-indent: 7px;
  letter-spacing: .7px;
  font-size: 1em;
}

form button {
  border: none;
  border-radius: 3px;
  background-color: #ceb3c5;
  color: #fff;
  font-size: 1.3em;
  height: 36px;
  letter-spacing: 1.2px;
  width: 40.5%;
  &:hover {
    background-color: #9c7e92
  }
}