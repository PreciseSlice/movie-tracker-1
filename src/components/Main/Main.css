.main {
  display: flex;
  padding: 15px 0;
  overflow-x: scroll; }

.main div {
  margin: 12px 0 0 0; }

.view-favorites {
  position: absolute;
  right: 20px;
  height: 25px;
  border: none;
  background-color: #ceb3c5;
  color: #fff;
  font-size: 0.9em;
  border-radius: 3px; }
  .view-favorites:hover {
    background-color: #9c7e92; }
