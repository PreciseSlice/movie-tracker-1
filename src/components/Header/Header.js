import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';
import { connect } from 'react-redux';
import { login } from '../../actions/index';
import PropTypes from 'prop-types';

export const Header = props => {
  const { loggedIn, logout } = props;
  const signInButtons = loggedIn ? (
    <button className="logout login-container" onClick={() => logout(false)}>
      Log Out
    </button>
  ) : (
    <div className="login-container">
      <NavLink to={{ pathname: '/sign-up' }}>
        <button className="sign-up">Sign Up</button>
      </NavLink>
      <NavLink to={{ pathname: '/login' }}>
        <button className="login">Login</button>
      </NavLink>
    </div>
  );

  return (
    <nav>
      {signInButtons}
      <NavLink to={{ pathname: '/' }}>
        <h1 className="title">Movie Tracker</h1>
      </NavLink>
    </nav>
  );
};

const mapStateToProps = state => ({
  loggedIn: state.loggedIn
});

const mapDispatchToProps = dispatch => ({
  logout: boolean => dispatch(login(boolean))
});

Header.propTypes = {
  loggedIn: PropTypes.bool,
  logout: PropTypes.bool
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
