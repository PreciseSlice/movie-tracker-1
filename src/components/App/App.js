import React, { Component } from 'react';
import Main from '../Main/Main';
import Header from '../Header/Header';
import Login from '../Login/Login';
import SignUp from '../SignUp/SignUp';
import { Switch, Route, Redirect } from 'react-router-dom';
import './App.css';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import PropTypes from 'prop-types';

export class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <Switch>
          <Route
            exact
            path="/"
            render={() => <Main movieData={this.props.movieData} />}
          />
          <Route
            path="/favorites"
            render={() =>
              this.props.loggedIn ? (
                <Main movieData={this.props.favorites} />
              ) : (
                <Main movieData={this.props.movieData} />
              )
            }
          />
          <Route
            path="/login"
            render={() =>
              this.props.loggedIn ? <Redirect to="/" /> : <Login />
            }
          />
          <Route
            path="/sign-up"
            render={() =>
              this.props.loggedIn ? <Redirect to="/" /> : <SignUp />
            }
          />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  loggedIn: state.loggedIn,
  movieData: state.movieData,
  favorites: state.favorites
});

App.propTypes = {
  loggedIn: PropTypes.bool.isRequired,

  movieData: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired,
      poster_path: PropTypes.string.isRequired,
      overview: PropTypes.string.isRequired,
      favorites: PropTypes.bool,
      release_date: PropTypes.string.isRequired,
      vote_average: PropTypes.number.isRequired
    })
  ).isRequired,

  favorites: PropTypes.array.isRequired
};

export default withRouter(connect(mapStateToProps, null)(App));
