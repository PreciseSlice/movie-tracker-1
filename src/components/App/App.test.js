import React, { Component } from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import { withRouter } from 'react-router';
import { shallow } from 'enzyme';
import { App, mapStateToProps } from './App'
import Main from '../Main/Main';
import Header from '../Header/Header';
import Login from '../Login/Login';
import SignUp from '../SignUp/SignUp';

describe('App', () => {
  let mockMovieData;
  let mockFavorites;

  beforeEach( () => {
    mockMovieData = [{
                      title: 'Coco',
                      movie_id: 354912,
                      poster_path: 'https://image.tmdb.jpg',
                      overview: 'Despite...',
                      release_date: '05/05/25',
                      vote_average: 10.0
                    }];
    mockFavorites = [{
                      id: 15,
                      movie_id: 395834,
                      overview: 'An FBI agent...',
                      poster_path: 'https://image.tmdb.org/',
                      release_date: '2017-08-03',
                      title: 'Wind River',
                      user_id: 2,
                      vote_average: '7.5'
                    }]    
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<App loggedIn={false} movieData={mockMovieData} favorites={mockFavorites} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should map the store correctly', () => {
    const mockStore = {
      loggedIn: false,
      movieData: mockMovieData,
      favorites: mockFavorites
    }
    const mapped = mapStateToProps(mockStore);
    expect(mapped.loggedIn).toEqual(mockStore.loggedIn)
  })
});
