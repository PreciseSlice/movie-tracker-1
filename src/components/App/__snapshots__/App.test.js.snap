// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className="App"
>
  <Connect(Component) />
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      path="/favorites"
      render={[Function]}
    />
    <Route
      path="/login"
      render={[Function]}
    />
    <Route
      path="/sign-up"
      render={[Function]}
    />
  </Switch>
</div>
`;
